Extract pose data

matharotheelf opened the pull request:
Added a class which manages the extraction of key pose data needed for the strecth classification process.

The feature data is:
- the displacement betwen the nose and the shoulder mid point (to measure head movement)
- the dispacement between the shoulder and elbows and elbows and hands for arm movement

This data is then converted into a single feature array which can be provided to a machine learning model for classification.

matharotheelf committed:
initial commit extract stretch pose data from keypoints

this is the key data required to classify stretches:

the stretch data includes:

- the displacement vector from the nose to the shoulder mid point
- the displacement vectors from each shoulder to their elbow
- the displacement vectors from each elbow to their hand

this simplifies the pose data for classifying the different stretches

matharotheelf committed:
comment description of data extraction code

matharotheelf committed:
remove debugging logs

matharotheelf committed:
visualise extracted data vectors for debugging

matharotheelf committed:
add function to convert extracted data into a features array for
classification

edennisCS commented on sketch.js:
```diff
@@ -10,6 +10,8 @@ let video;
 let bodyPose;
 let poses = [];
 let connections;
+let timeAccumalator = 0;
```
I think better that accumalator should be spelt like accumulator

edennisCS commented on sketch.js:
```diff
@@ -34,6 +36,13 @@ function draw() {
 
+  timeAccumalator += deltaTime;
+
+  if(timeAccumalator >= poseSampleInterval && poses.length == 1) {
```
maybe strict equality is better === though unlikely to be any issues

edennisCS commented on sketch.js:
```diff
@@ -34,6 +36,13 @@ function draw() {
   // Draw the webcam video
```
Draw function looks good though could work as multiple sub functions to separate logic (as I saw it has three different stages) not critical

edennisCS commented:
There are some places with missing semi colons I think it is fine for now though as they are not in critical areas. Maybe can auto format it later or something

Commit:
Add semi colons to lint

Commit:
Add eslint to project for linting

matharotheelf commented:
> There are some places with missing semi colons I think it is fine for now though as they are not in critical areas. Maybe can auto format it later or something

I added eslint and fixed the missing semi colons!

edennisCS reviewed (commented):
Forgot to push these as review, I also made these suggestions :P
